doc/languages-frameworks/rust.section.md

```diff
@@ -477,7 +485,29 @@ in the `~/.config/nixpkgs/overlays` directory.
     $ mkdir -p ~/.config/nixpkgs/overlays
     $ ln -s $(pwd)/nixpkgs-mozilla/rust-overlay.nix ~/.config/nixpkgs/overlays/rust-overlay.nix
 
-The latest version can be installed with the following command:
+### Declarative rust overlay installation
+
+Add the following to your `configuration.nix` (or similar):
+
+```
+  nixpkgs = {
+    overlays = [
+      (import (builtins.fetchTarball https://github.com/mozilla/nixpkgs-mozilla/archive/master.tar.gz))
+      # Further overlays go here
+    ];
+  };
+```
+
+Note that this will fetch the latest overlay version when rebuilding your system.
+
+### Rust overlay usage
+
+The overlay contains different versions of the rust toolchain,
+such as attribute sets `latest.rustChannels.stable`, `latest.rustChannels.nightly`, and a function `(rustChannelOf { date = "2018-04-11"; channel = "nightly"; })`.
```

Comment:
I might also mention that `rustChannelOf` can point to a `rust-toolchain` file, which is especially useful in `shell.nix`

Reply:
Yes, I didn't talk about `shell.nix` since I'm not using it. I can add some information if you can check it.

Reply:
How about this?